core/ui/compose/theme/src/main/java/app/k9mail/core/ui/compose/theme/IconsWithBaseline.kt

```diff
@@ -0,0 +1,16 @@
+package app.k9mail.core.ui.compose.theme
+
+import androidx.compose.material.icons.filled.Warning
+import androidx.compose.ui.unit.dp
+import app.k9mail.core.ui.compose.common.image.ImageWithBaseline
+import androidx.compose.material.icons.Icons as MaterialIcons
+
+// We're using "by lazy" so not all icons are loaded into memory as soon as a nested object is accessed. But once a
+// property is accessed we want to retain the `ImageWithBaseline` instance.
+object IconsWithBaseline {
+    object Filled {
+        val warning: ImageWithBaseline by lazy {
+            ImageWithBaseline(image = MaterialIcons.Filled.Warning, baseline = 21.dp)
```

Comment:
Where is `baseline = 21.dp` derived from? Is it universally usable or bound to a specific context?

Reply:
It's the manually derived baseline for the default size of that particular icon. It's the offset for the lower edge of the triangle. The icon itself contains some empty space at the bottom, so we don't want to align to the bottom edge of it.

![image](https://github.com/thundernest/k-9/assets/218061/127e9855-01f7-40ea-bbf7-215a725a5859)